File: Task1_2.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0));
    
    int N;
    do {
        cout << "Введите размер матрицы: ";
        cin >> N;
    } while (N % 2 != 0);
    
    // Инициализация матрицы
    vector<vector<int>> matrix(N, vector<int>(N));
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            matrix[i][j] = rand() % 21 + 10; // [10, 30]
        }
    }
    
    // Вывод исходной матрицы
    cout << "Исходная матрица:" << endl;
    for (const auto& row : matrix) {
        for (int num : row) {
            cout << num << "\t";
        }
        cout << endl;
    }
    
    // Размер четверти
    int quarter = N / 2;
    
    // Левая нижняя четверть
    for (int i = quarter; i < N; ++i) {
        for (int j = 0; j < quarter; ++j) {
            matrix[i][j] = 0;
        }
    }
    
    // Правая нижняя четверть
    for (int i = quarter; i < N; ++i) {
        for (int j = quarter; j < N; ++j) {
            matrix[i][j] = 10;
        }
    }
    
    // Меняем местами оставшиеся четверти
    for (int i = 0; i < quarter; ++i) {
        for (int j = 0; j < quarter; ++j) {
            swap(matrix[i][j], matrix[i][j + quarter]);
        }
    }
    
    // Вывод измененной матрицы
    cout << "\nИзмененная матрица:" << endl;
    for (const auto& row : matrix) {
        for (int num : row) {
            cout << num << "\t";
        }
        cout << endl;
    }
    return 0;
}

File: Task1_2.rs
use rand::Rng;
use std::io;

fn main() {
    let mut rng = rand::thread_rng();
    
    // Ввод размера матрицы (чётного числа)
    let n = loop {
        println!("Введите размер матрицы: ");
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Ошибка чтения");
        let n: usize = match input.trim().parse() {
            Ok(num) if num % 2 == 0 => num,
            _ => {
                println!("Размер должен быть чётным числом!");
                continue;
            }
        };
        break n;
    };
    
    // Инициализация матрицы случайными числами [10, 30]
    let mut matrix = vec![vec![0; n]; n];
    for i in 0..n {
        for j in 0..n {
            matrix[i][j] = rng.gen_range(10..=30);
        }
    }
    
    // Вывод исходной матрицы
    println!("Исходная матрица:");
    for row in &matrix {
        for num in row {
            print!("{}\t", num);
        }
        println!();
    }
    
    let quarter = n / 2;
    
    // Обработка левой нижней четверти (заполняем нулями)
    for i in quarter..n {
        for j in 0..quarter {
            matrix[i][j] = 0;
        }
    }
    
    // Обработка правой нижней четверти (заполняем десятками)
    for i in quarter..n {
        for j in quarter..n {
            matrix[i][j] = 10;
        }
    }
    
    // Обмен верхних четвертей местами
    for i in 0..quarter {
        for j in 0..quarter {
            matrix[i].swap(j, j + quarter);
        }
    }
    
    // Вывод изменённой матрицы
    println!("\nИзмененная матрица:");
    for row in &matrix {
        for num in row {
            print!("{}\t", num);
        }
        println!();
    }
}
